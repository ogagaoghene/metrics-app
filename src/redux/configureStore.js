import { legacy_createStore as createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import reduContinent from './continent';
import reduCountry from './country';

const rootReducer = combineReducers({
  reduContinent,
  reduCountry,
});

const store = createStore(
  rootReducer,
  applyMiddleware(logger, thunk),
);

export default store;
