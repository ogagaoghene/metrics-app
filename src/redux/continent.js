import { getAllContinents } from '../components/Data/Data';

// Actions types
const GET_DATA = 'metrics-webapp/continent/GET_DATA';

// Reducer
export default function reducer(state = [], action = {}) {
  switch (action.type) {
    case GET_DATA:
      return action.payload;

    default:
      return state;
  }
}

// Action Creators
export const getAllData = async (dispatch) => {
  const continents = await getAllContinents();
  dispatch({ type: GET_DATA, payload: continents });
};
