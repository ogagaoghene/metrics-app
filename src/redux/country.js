import { getApiDataCountry } from '../components/Data/Data';

// Actions types
const GET_COUNTRY_INFO = 'metrics-webapp/country/GET_COUNTRY_INFO';

// Reducer
export default function reducer(state = [], action = {}) {
  switch (action.type) {
    case GET_COUNTRY_INFO:
      return action.payload;

    default:
      return state;
  }
}

// Action Creators
export const getCountryInfo = (country) => async (dispatch) => {
  const countryInfo = await getApiDataCountry(`${country}`);
  // console.log(countryInfo);
  dispatch({ type: GET_COUNTRY_INFO, payload: countryInfo });
};
