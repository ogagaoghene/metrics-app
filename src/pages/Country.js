import { useParams } from 'react-router-dom';
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getAllData } from '../redux/continent';
import { getCountryInfo } from '../redux/country';
import {
  Container, List, Button, InfoCtry,
} from '../components/Styles/CountriesStyle';

function Country() {
  const dispatch = useDispatch();
  const { continent } = useParams();
  const continents = useSelector((state) => state.reduContinent) || [];
  const countryInfo = useSelector((state) => state.reduCountry);
  useEffect(() => {
    dispatch(getAllData);
  }, []);

  const filtercontinent = continents.filter((item) => (item.id === continent))[0];
  const { countries } = filtercontinent || [];
  const [country, setCountry] = useState();

  return (
    <Container>
      <h1>{continent}</h1>
      <List name="country" onChange={(e) => setCountry(e.target.value)}>
        <option value="" key="country">Country...</option>
        {countries && countries.map((ctry) => <option value={`${ctry}`} key={ctry}>{ctry}</option>)}
      </List>
      <Button type="button" onClick={() => dispatch(getCountryInfo(country))}>Show Info</Button>
      {country && (country === countryInfo.id)
        ? (
          <InfoCtry>
            <li>
              <h3>{countryInfo.id}</h3>
            </li>
            <li><img src={countryInfo.flag} alt="flag" width={150} /></li>
            <br />
            <li>
              <strong>Cases:</strong>
              {' '}
              {countryInfo.cases}
            </li>
            <br />
            <li>
              <strong>Active:</strong>
              {' '}
              {countryInfo.active}
            </li>
            <br />
            <li>
              <strong>Critical:</strong>
              {' '}
              {countryInfo.critical}
            </li>
            <br />
            <li>
              <strong>Recovered:</strong>
              {' '}
              {countryInfo.recovered}
            </li>
            <br />
            <li>
              <strong>Deaths:</strong>
              {' '}
              {countryInfo.deaths}
            </li>
            <br />
            <li>
              <strong>Population:</strong>
              {' '}
              {countryInfo.population}
            </li>
          </InfoCtry>
        )
        : 'CHOOSE A COUNTRY'}
    </Container>
  );
}

export default Country;
