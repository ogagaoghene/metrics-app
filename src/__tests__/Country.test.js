import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Country from '../pages/Country';
import store from '../redux/configureStore';
import { getCountryInfo } from '../redux/country';

describe('Test Country page', () => {
  test('Should match the snapshot', () => {
    const dom = render(
      <Provider store={store}>
        <Router>
          <Country />
        </Router>
      </Provider>,
    );
    expect(dom).toMatchSnapshot();
  });

  test('Should return an object with the info of the country', async () => {
    const countries = await getCountryInfo('Moroco');
    expect(countries.length).toBe(1);
  });
});
