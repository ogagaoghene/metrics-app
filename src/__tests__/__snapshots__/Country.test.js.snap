// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Country page Should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ cBhKuY"
      >
        <h1 />
        <select
          class="sc-gsnTZi kpDwFE"
          name="country"
        >
          <option
            value=""
          >
            Country...
          </option>
        </select>
        <button
          class="sc-dkzDqf kCzbPk"
          type="button"
        >
          Show Info
        </button>
        CHOOSE A COUNTRY
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ cBhKuY"
    >
      <h1 />
      <select
        class="sc-gsnTZi kpDwFE"
        name="country"
      >
        <option
          value=""
        >
          Country...
        </option>
      </select>
      <button
        class="sc-dkzDqf kCzbPk"
        type="button"
      >
        Show Info
      </button>
      CHOOSE A COUNTRY
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
